Assets/BossRoom/Scripts/Shared/Net/ConnectionManagement/ClientGameNetPortal.cs

```diff
@@ -177,40 +178,46 @@ public void StartClient(string ipaddress, int port)
                     throw new ArgumentOutOfRangeException(nameof(chosenTransport));
             }
 
-            ConnectClient();
+            ConnectClient(null);
         }
 
-        public async void StartClientUnityRelayModeAsync(string joinCode, Action<string> onFailure)
+        public void StartClientUnityRelayModeAsync(string joinCode, Action<string> onFailure)
         {
-            var utp = (UnityTransport)NetworkManager.Singleton.gameObject.GetComponent<TransportPicker>().UnityRelayTransport;
+            m_Portal.OnlineMode = OnlineMode.UnityRelay;
+            m_JoinCode = joinCode;
+            var utp = NetworkManager.Singleton.gameObject.GetComponent<TransportPicker>().UnityRelayTransport;
             NetworkManager.Singleton.NetworkConfig.NetworkTransport = utp;
 
             Debug.Log($"Setting Unity Relay client with join code {joinCode}");
 
-            try
-            {
-                var clientRelayUtilityTask =  UnityRelayUtilities.JoinRelayServerFromJoinCode(joinCode);
-                await clientRelayUtilityTask;
-                var (ipv4Address, port, allocationIdBytes, connectionData, hostConnectionData, key) = clientRelayUtilityTask.Result;
+            ConnectClient(onFailure);
+        }
 
-                m_LobbyServiceFacade.UpdatePlayerRelayInfoAsync(allocationIdBytes.ToString(), joinCode, null, null);
-                utp.SetClientRelayData(ipv4Address, port, allocationIdBytes, key, connectionData, hostConnectionData, isSecure: true);
-            }
-            catch (Exception e)
+        async void ConnectClient(Action<string> onFailure)
```

Comment:
onFailure isn't used?

Reply:
It's used when catching an exception while starting with relay mode, but this is not needed for this PR, I'll remove it to make things simpler.